test/specs/commands/createTransactionCastVote.js

```diff
@@ -0,0 +1,241 @@
+								Given('an empty options object', given.anEmptyOptionsObject, () => {
+									When('the action is called with the options', when.theActionIsCalledWithTheOptions, () => {
+										Then('it should reject with message "At least one of vote and/or unvote options must be provided."', then.itShouldRejectWithMessage);
+									});
+								});
+								Given('an options object with vote set to "file:path/to/votes.txt" and passphrase set to "passphraseSource" and second passphrase set to "secondPassphraseSource"', given.anOptionsObjectWithVoteSetToPassphraseSetToAndSecondPassphraseSetTo, () => {
+									Given('an error "Unknown data source type. Must be one of `file`, or `stdin`." occurs retrieving the inputs from their sources', given.anErrorOccursRetrievingTheInputsFromTheirSources, () => {
+										When('the action is called with the amount, the address and the options', when.theActionIsCalledWithTheAmountTheAddressAndTheOptions, () => {
+											Then('it should reject with the error message', then.itShouldRejectWithTheErrorMessage);
+										});
+									});
+								});
+								Given('the passphrase can be retrieved from its source', given.thePassphraseCanBeRetrievedFromItsSource, () => {
+									Given('an options object with vote set to "file:path/to/votes.txt" and unvote set to "file:path/to/votes.txt"', given.anOptionsObjectWithVoteSetToAndUnvoteSetTo, () => {
+										When('the action is called with the options', when.theActionIsCalledWithTheOptions, () => {
+											Then('it should reject with message "Vote and unvote sources must not be the same."', then.itShouldRejectWithMessage);
+										});
+									});
+									Given('an options object with vote set to "12345678123456781234567812345678123456781234567812345678123456gg"', given.anOptionsObjectWithVoteSetTo, () => {
+										When('the action is called with the options', when.theActionIsCalledWithTheOptions, () => {
+											Then('it should reject with validation error and message "Public key 12345678123456781234567812345678123456781234567812345678123456gg bytes length differs from the expected 32 bytes for a public key."', then.itShouldRejectWithValidationErrorAndMessage);
+										});
+									});
+									Given('an options object with vote set to "215b667a32a5cd51a94c9c2046c11,922fbfdd596fa78269bbcadc67ec"', given.anOptionsObjectWithVoteSetTo, () => {
+										When('the action is called with the options', when.theActionIsCalledWithTheOptions, () => {
+											Then('it should reject with validation error and message "Error processing public key 215b667a32a5cd51a94c9c2046c11: Invalid hex string."', then.itShouldRejectWithValidationErrorAndMessage);
+										});
+									});
+									Given('an options object with unvote set to "215b667a32a5cd51a94c9c2046c11,922fbfdd596fa78269bbcadc67ec"', given.anOptionsObjectWithUnvoteSetTo, () => {
```

Comment:
Why do we check unvote for this case but not for the others?

Reply:
vote is also checked directly above.

Reply:
Yes, here we check both, but in other cases we check only vote.